### DISCUSSION TITLE
What's a good way to ignore errors in capfs?

### DISCUSSION THREAD

**Discussion by NightMachinery** [minad/cape#110] [Q&A] [answered]
In my org files, I have:
```
completion-at-point-functions is a variable defined in minibuffer.el.

Value in #<buffer x.org>
(jupyter-org-completion-at-point
 pcomplete-completions-at-point
 t)
```
`jupyter-org-completion-at-point` sometimes throws an error when a remote kernel is not connected (etc.). E.g.,:
```
if: Symbol’s function definition is void: org-babel-jupyter-language-p
```

I still like to see my other capf completions such as `cape-dabbrev`. What's the best way to do this?

Here are my global capfs:
```
(setq-default completion-at-point-functions
                (list
                 #'cape-file

                 ;; `cape-capf-super' is experimental.
                 ;; [https://github.com/minad/cape#super-capf---merging-multiple-capfs]
                 ;; [https://github.com/minad/cape#capf-buster---cache-busting]
                 (cape-capf-super
                  #'cape-dabbrev
                  #'cape-keyword)))
```

**Comment by minad** [comment]
Hi @NightMachinery! 

The Capf transformer you are looking for is listed in the README, but perhaps not that easy to find. ;) Generally I would recommend to fix the real error instead of hiding the symptoms.

(Please slow a bit down with creating new questions/issues and study the READMEs and maybe also the code. You should be able to find many answers there. If every user would pose ten questions a day, I would be flooded quickly. ;) )

**Reply by NightMachinery** [reply]
Thanks, sorry for being a bother. Feel free to just reply "too busy now" for any issue/question I create.

I took another look at the readme, and `cape-capf-accept-all` and `cape-capf-debug` look somewhat related, though I couldn't really understand what `cape-capf-accept-all` does.

I ended up using this hack:
```lisp
(defun night/prioritize-global-capf ()
  "Moves global capfs to the beginning of capf list, so that local capfs won't break the global capfs."
  (interactive)
  (setq-local
   completion-at-point-functions
   (cl-remove-duplicates (-filter
                          ;; Removing =t= to avoid using the global capfs to avoid repetition:
                          (lambda (x)
                            (not (equal x t)))
                          (-concat
                           (default-value 'completion-at-point-functions)
                           completion-at-point-functions)))))
```

**Reply by minad** [reply, marked as answer]
Maybe it is not sufficiently obvious. cape-capf-accept-all is completely unrelated. The solution is cape-capf-silent. See https://github.com/minad/cape/commit/7353489c38622b1f1698ecfcc8d19ec7bfd71e1b.
